src/main/java/net/minecraftforge/common/util/ForgeSoundType.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2020.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.util;
+
+import net.minecraft.block.SoundType;
+import net.minecraft.util.SoundEvent;
+
+import javax.annotation.Nonnull;
+import java.util.function.Supplier;
+
+public class ForgeSoundType extends SoundType
```

Comment:
Might be worth throwing some java docs on this class that gives a brief summary of why this exists (given you wrote a very detailed thing for the PR explaining it you can probably just copy paste parts of that here)

Reply:
Added a condensed/summarized version of the PR description as javadocs to the `ForegSoundType` class. This should give modders more insight into why it exists.
(I thought about linking this PR's URL in the `@see` section, but decided against it for now. If modders are curious, they could always `git blame`/search on GitHub to find the commit and this PR).